Amsterdam-Music-Lab/MUSCLE
# Config: Configure Sentry for backend

This PR adds a basic Sentry integration to the Django backend of the MUSCLE platform. I've verified it locally and it *works*, but some questions are still to be answered before we should merge this PR.

## Questions
* For which environments should we configure Sentry?
  * Production is a given, so yes.
  * Acceptance: Could come in handy if we want to spot errors & bugs early before going to production, but can eat into the error rate limit of the free tier of Sentry 😅 
  * Local develop env: It is not very common to configure Sentry for a local environment as you (the developer) are the only user so you have easy access to logs and verbose error messages anyway.
* Then there's the question of avoiding to send sensitive information to Sentry. This [is configurable](https://docs.sentry.io/platforms/javascript/data-management/sensitive-data/), but how should we approach it? We can merge this PR and see what information currently gets sent to Sentry in Acceptance and use a follow up PR that configures the *scrubbing* of the data. Or we can already try to do that locally after which I will update this PR to also scrub data from sensitive information.

## How to test this PR?
* Pull and checkout this branch
* Configure the `SENTRY_DSN` key (ask @drikusroor for it)
* Rebuild the Docker container (there are new python packages to be installed)
* Uncomment the `sentry-debug` url in `backend/aml/urls.py` and navigate towards that route (`http://localhost:8000/sentry-debug`) to trigger an error that gets sent to Sentry.
* Go to [Sentry.io](https://sentry.io), log in (again, ask @drikusroor for credentials) and see if your error has (re-)appeared. The sample rate is currently set to `0.2` (see `backend/aml/base_settings.py`) so only 20% of the errors will get reported, so you might have to refresh a couple of times.

Related to #636

Resolves #629
